pkg/storaged/resize.jsx

```diff
@@ -203,16 +212,54 @@ export function get_resize_info(client, block, to_fit) {
     return { info, shrink_excuse, grow_excuse };
 }
 
-export function grow_dialog(client, lvol, info, to_fit) {
-    const block = client.lvols_block[lvol.path];
-    const vgroup = client.vgroups[lvol.VolumeGroup];
-    const pool = client.lvols[lvol.ThinPool];
+export function free_space_after_part(client, part) {
+    const parts = get_partitions(client, client.blocks[part.Table]);
+
+    function find_it(parts) {
+        for (const p of parts) {
+            if (p.type == "free" && p.start == part.Offset + part.Size)
+                return p.size;
+            if (p.type == "container") {
+                const s = find_it(p.partitions);
+                if (s)
+                    return s;
+            }
+        }
+        return false;
+    }
+
+    return find_it(parts) || 0;
+}
+
+export function grow_dialog(client, lvol_or_part, info, to_fit) {
+    let title, block, name, orig_size, max_size, allow_infinite, round_size;
+
+    if (lvol_or_part.iface == "org.freedesktop.UDisks2.LogicalVolume") {
+        const vgroup = client.vgroups[lvol_or_part.VolumeGroup];
+        const pool = client.lvols[lvol_or_part.ThinPool];
+
+        title = _("Grow logical volume");
+        block = client.lvols_block[lvol_or_part.path];
+        name = lvol_or_part.Name;
+        orig_size = lvol_or_part.Size;
+        max_size = pool ? pool.Size * 3 : lvol_or_part.Size + vgroup.FreeSize;
+        allow_infinite = !!pool;
+        round_size = vgroup.ExtentSize;
+    } else {
+        title = _("Grow partition");
+        block = client.blocks[lvol_or_part.path];
+        name = block_name(block);
+        orig_size = lvol_or_part.Size;
+        max_size = lvol_or_part.Size + free_space_after_part(client, lvol_or_part);
+        allow_infinite = false;
+        round_size = 1024 * 1024;
+    }
 
     const usage = get_active_usage(client, block && info.grow_needs_unmount ? block.path : null, _("grow"));
 
     if (usage.Blocking) {
         dialog_open({
-            Title: cockpit.format(_("$0 is in use"), lvol.Name),
+            Title: cockpit.format(_("$0 is in use"), name),
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19385-20230928-111839-72431268-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)